api/v1alpha4/conversion.go

```diff
@@ -272,13 +273,28 @@ func (dst *MachineDeploymentList) ConvertFrom(srcRaw conversion.Hub) error {
 func (src *MachineHealthCheck) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*clusterv1.MachineHealthCheck)
 
-	return Convert_v1alpha4_MachineHealthCheck_To_v1beta1_MachineHealthCheck(src, dst, nil)
+	if err := Convert_v1alpha4_MachineHealthCheck_To_v1beta1_MachineHealthCheck(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &clusterv1.MachineHealthCheck{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	return nil
```

Comment:
Given that we rolled back changes to MHC + we don't restore anything we should revert both changes to MHC conversion